packages/suite/src/views/wallet/anonymize/components/CoinjoinConfirmation.tsx

```diff
@@ -130,20 +132,20 @@ export const CoinjoinConfirmation = ({ account }: CoinjoinConfirmationProps) =>
 
     const maxRounds = getMaxRounds(targetAnonymity, anonymitySet);
     const allAnonymized = notAnonymized === '0';
-    const deviceIsLockedOrDisconnected = isLocked(false);
 
     const isDisabled =
         !termsConfirmed ||
         allAnonymized ||
-        deviceIsLockedOrDisconnected ||
+        !deivce?.connected ||
+        isLocked(true) ||
         isCoinjoinBlockedByTor ||
         isCoinjoinDisabledByFeatureFlag;
 
     const getButtonTooltipMessage = () => {
         if (isCoinjoinDisabledByFeatureFlag && featureMessageContent) {
             return featureMessageContent;
         }
-        if (deviceIsLockedOrDisconnected) {
+        if (!deivce?.connected) {
```

Comment:
Device can be locked for some other reason so I'm not sure about this solution. Isn't there some way to add "loading" state for this button?